sponsorship_compassion/model/contracts.py

```diff
@@ -191,266 +52,88 @@ def _get_standard_lines(self, cr, uid, context=None):
-            else:
-                next_invoice_date = current_date.replace(day=1)
-            payment_term = contract_group.payment_term_id.name
-        else:
-            next_invoice_date = current_date.replace(day=1)
-            payment_term = ''
-
-        if current_date.day > 15 or (
-                payment_term in ('LSV', 'Postfinance') and not is_active):
-            next_invoice_date = next_invoice_date + relativedelta(months=+1)
-        res['value'] = {'next_invoice_date': next_invoice_date.strftime(DF)}
-        return res
-
-    def contract_waiting(self, cr, uid, ids, context=None):
-        for contract in self.browse(cr, uid, ids, {'lang': 'en_US'}):
-            payment_term = contract.group_id.payment_term_id.name
-            if 'LSV' in payment_term or 'Postfinance' in payment_term:
-                # Recompute next_invoice_date
-                today = datetime.today()
-                next_invoice_date = datetime.strptime(
-                    contract.next_invoice_date, DF).replace(month=today.month,
-                                                            year=today.year)
-                if today.day > 15 and next_invoice_date.day < 15:
-                    next_invoice_date = next_invoice_date + relativedelta(
-                        months=+1)
-                contract.write({
-                    'next_invoice_date': next_invoice_date.strftime(DF)})
-
-            # Check that a child is selected for Sponsorship product
-            if not contract.child_id:
-                for line in contract.contract_line_ids:
-                    if line.product_id.name == 'Sponsorship' or \
-                            'LDP' in line.product_id.name:
-                        raise orm.except_orm(
-                            _("Please select a child"),
-                            _("You should select a child if you "
-                              "make a new sponsorship!"))
-
-        self.write(cr, uid, ids, {'state': 'waiting'}, context)
+    ##############################
+    #      CALLBACKS FOR GP      #
+    ##############################
+    def force_validation(self, cr, uid, contract_id, context=None):
+        """ Used to transition draft sponsorships in waiting state
+        when exported from GP. """
+        wf_service = netsvc.LocalService('workflow')
+        logger.info("Contract " + str(contract_id) + " validated.")
+        wf_service.trg_validate(uid, self._name, contract_id,
+                                'contract_validated', cr)
         return True
 
-    def contract_cancelled(self, cr, uid, ids, context=None):
-        self.write(cr, uid, ids, {'state': 'cancelled'}, context)
-        # Remove the sponsor of the child
-        for contract in self.browse(cr, uid, ids, context):
-            if contract.child_id:
-                contract.child_id.write({'sponsor_id': False})
+    def force_activation(self, cr, uid, contract_id, context=None):

```

Comment:
This method should be moved to contract_compassion as it is used in activate_contract_wizard.